rasa/utils/tensorflow/environment.py

```diff
@@ -0,0 +1,112 @@
+import logging
+import os
+from typing import Text, Tuple, Dict
+import warnings
+from rasa.constants import (
+    ENV_GPU_CONFIG,
+    ENV_CPU_INTER_OP_CONFIG,
+    ENV_CPU_INTRA_OP_CONFIG,
+)
+from tensorflow import config as tf_config
+
+logger = logging.getLogger(__name__)
+
+
+def setup_gpu_environment() -> None:
+    """Set configuration for a GPU environment based on the environment variable set"""
+
+    gpu_memory_config = os.getenv(ENV_GPU_CONFIG)
+    if gpu_memory_config:
+        parsed_gpu_config = parse_gpu_config(gpu_memory_config)
+        physical_gpus = tf_config.list_physical_devices("GPU")
+
+        # Logic taken from https://www.tensorflow.org/guide/gpu
+        if physical_gpus:
+            for gpu_id, gpu_id_memory in parsed_gpu_config.items():
+                try:
+                    tf_config.experimental.set_virtual_device_configuration(
+                        physical_gpus[gpu_id],
+                        [
+                            tf_config.experimental.VirtualDeviceConfiguration(
+                                memory_limit=gpu_id_memory
+                            )
+                        ],
+                    )
+
+                except RuntimeError:
+                    # Add a helper explanation where the error comes from
+                    raise RuntimeError(
+                        "Error while setting up tensorflow environment. "
+                        "Virtual devices must be set before GPUs have been initialized"
+                    )
+
+        else:
+            warnings.warn(
```

Comment:
should we use our internal method `raise_warning`?

Reply:
yes please!

Reply:
Done